torchgen/model.py

```diff
@@ -489,7 +501,11 @@ def from_yaml(
         structured_delegate_s = e.pop("structured_delegate", None)
         assert structured_delegate_s is None or isinstance(
             structured_delegate_s, str
-        ), f"not a str: {structured_delegate}"
+        ), f"not a str: {structured_delegate_s}"
+        assert structured_delegate_s is None or "::" not in structured_delegate_s, (
```

Comment:
oh if we're gonna add this check, we should probably do the same for `structured_inherits`

Reply:
I don't think i see this check?

Reply:
Oops yeah I missed it